### DISCUSSION TITLE
How do I resolve IMigrationRunner ?

### DISCUSSION THREAD

**Discussion by goto3** [fluentmigrator/fluentmigrator#1540] [General] [answered]
Hello, I can't make it work on with my Web Api using .NET 6 and postgreSQL, I get this error in Program.cs:

![image](https://user-images.githubusercontent.com/16456164/141696781-3717179c-f5b1-4e5f-ac8b-d4f0cc3cc29c.png)

this is how I add the FluentMigrator to the services:

![image](https://user-images.githubusercontent.com/16456164/141696796-2a98704c-367c-4d5f-a960-d115cb2c8b1a.png)

am I missing something?

**Comment by goto3** [comment, marked as answer]
Solution:

![image](https://user-images.githubusercontent.com/16456164/141700092-afd22171-868f-4e6a-95cb-70d1b6abd1fc.png)

**Reply by fubar-coder** [reply]
The migration runner is indeed a scoped service.

**Reply by jzabroski** [reply]
@goto3 I changed your question title as I don't want people to land here from a Google Search and think the original title was misleading, but understand why you titled it the way you did.  Glad you found the solution.

**Reply by goto3** [reply]
thanks to both of you for making this awesome tool
